test: properly unset recovery_leader in the recovery procedure tests

patjed41 opened the pull request:
After changing the type of the `recovery_leader` config option from
`sstring` to `UUID` in #25032, setting `recovery_leader` to an empty
string became an incorrect way to unset it. The following error started
to appear in the recovery procedure tests:
```
init - marshaling error: UUID string size mismatch: '' : recovery_leader
```
We unset `recovery_leader` properly in this PR. To do it, we introduce
a simple way to remove config options in tests.

Backport is unneeded. This error was harmless, and Scylla ignored
`recovery_leader` after logging the error as expected by the tests.

patjed41 commented:
@enaydanov Can you review the first two commits?

patjed41 commented on test/pylib/manager_client.py:
```diff
@@ -531,9 +531,15 @@ async def server_update_config(self,
-            if value is None:
-                raise RuntimeError("`value` is required if `key` is not None")
             if config_options is not None:
                 raise RuntimeError("`key: value` pair and `config_options` dict can't be used simultaneously")
```
> The validation logic has been removed but there's no handling for the case where key is provided but value is not explicitly passed (defaulting to None).

I don't see a good way to fix this. The only reasonable default values is `None` and we should have a default value, or else the caller would have to explicitly pass `None` when using `config_options`.

> This could lead to unintended config option removal when the caller meant to skip providing a value entirely.

There is no reason why a caller would not provide a value other than removing the config option.

enaydanov replied:
Personally, I prefer to have a separate explicit method for an option removal. Like `server_remove_config_option`.

Corresponding method in `ScyllaServer` class can look like:

```python
    def remove_config_option(self, key: str) -> None:
        """Remove an option from conf/scylla.yaml.

        If we're running, reload the config with a SIGHUP.
        """
        self.config.pop(key, None)  # don't fail if there is no such option in the config
        self._write_config_file()
        if self.cmd:
            self.cmd.send_signal(signal.SIGHUP)
```

patjed41 commented:
Docs with the `None` -> `null` conversion: https://pyyaml.org/wiki/PyYAMLDocumentation#yaml-tags-and-python-types

patjed41 commented:
v2:
- added `ManagerClient.server_remove_config_option` that removes the config option instead of setting it to `null`
- reverted the change in `ManagerClient.server_update_config`
- used `ManagerClient.server_remove_config_option` in the recovery procedure tests
- updated the cover letter and commit messages to reflect the changes above

patjed41 commented:
I don't know why CI failed, maybe https://github.com/scylladb/scylla-pkg/issues/5337
